config/routes.rb

```diff
@@ -246,7 +246,11 @@
 
     resources :permissions, :only => [:index]
 
-    resources :auth_source_ldaps, :except => [:show]
+    resources :auth_source_ldaps, :except => [:show] do
+      collection do
+        put 'test_connection'
```

Comment:
The regular controller works different, i.e it create a temp ldap object and test it, so I chose  put method w/ params.
The API controller has no params but ldap id/name for testing it.